### PR TITLE
Xinputcheck improvements from #2149

### PR DESCRIPTION
**Pull request by mkrautz**

### CODE REVIEW

**Comment by hacst** [comment]
LGTM.

**Comment by hacst** [comment]
Eh wait. I take that back ;) The clear cache leaks memory.

File: 3rdparty/xinputcheck-src/xinputcheck.cpp [hunk 1]
```diff
@@ -111,3 +140,8 @@ SDL_IsXInputDevice(const GUID* pGuidProductFromDirectInput)
 bool XInputCheck_IsGuidProductXInputDevice(const GUID *pGuidProductFromDirectInput) {
     return SDL_IsXInputDevice(pGuidProductFromDirectInput);
 }
+
+void XInputCheck_ClearDeviceCache() {
+    SDL_RawDevList = NULL;
```
Comment (hacst) [inline 3rdparty/xinputcheck-src/xinputcheck.cpp]: This leaks the cache memory.

Comment (mkrautz) [inline 3rdparty/xinputcheck-src/xinputcheck.cpp]: Yep. D'oh.

**Comment by hacst** [comment]
LGTM.

**Comment by mkrautz** [comment]
Thanks, doing a local build check before landing...

**Comment by mkrautz** [comment]
Landed.

**Pull request merged**